### PR TITLE
chore: add osv scanner

### PR DESCRIPTION
**Pull request by npalm**
Add OSV scanner to check for security issues.

### CODE REVIEW

File: .github/workflows/osv-scanner.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# A sample workflow which sets up periodic OSV-Scanner scanning for vulnerabilities,
+# in addition to a PR check which fails if new vulnerabilities are introduced.
+#
+# For more examples and options, including how to ignore specific vulnerabilities,
+# see https://google.github.io/osv-scanner/github-action/
```
Comment (rjaegers) [inline .github/workflows/osv-scanner.yml:10]: I would remove these comments, I don't think they add much. Especially the first paragraph which can be added to almost all workflows.

File: .github/workflows/osv-scanner.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    branches: [ "main" ]
+  schedule:
+    - cron: '40 15 * * 1'
+  push:
+    branches: [ "main" ]
+
+permissions:
+  # Require writing security events to upload SARIF file to security tab
+  security-events: write
+  # Read commit contents
```
Comment (rjaegers) [inline .github/workflows/osv-scanner.yml:27]: Remove comment that is obvious.

File: .github/workflows/osv-scanner.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      # Example of specifying custom arguments
+      scan-args: |-
+        -r
+        --skip-git
+        ./
+  scan-pr:
+    if: ${{ github.event_name == 'pull_request' || github.event_name == 'merge_group' }}
+    uses: "google/osv-scanner-action/.github/workflows/osv-scanner-reusable-pr.yml@1f1242919d8a60496dd1874b24b62b2370ed4c78" # v1.7.1
+    with:
+      # Example of specifying custom arguments
```
Comment (rjaegers) [inline .github/workflows/osv-scanner.yml:44]: Remove comments.

**Review by rjaegers — DISMISSED** [review]

File: .github/workflows/osv-scanner.yml (hunk 1 above)
Comment (npalm) [reply .github/workflows/osv-scanner.yml:10]: nice that generator :)

**Review by rjaegers — APPROVED** [review]

**Pull request merged**